tools/autograd/templates/ViewFuncs.cpp

```diff
@@ -0,0 +1,69 @@
+#include "torch/csrc/autograd/generated/ViewFuncs.h"
+
+// ${generated_comment}
+
+using at::Tensor;
+using at::Scalar;
+using at::IntArrayRef;
+using at::TensorList;
+
+namespace torch::autograd {
+
+ChainedViewFunc::ChainedViewFunc(
+  const std::shared_ptr<ViewFunc>& first_,
+  const std::shared_ptr<ViewFunc>& second_)
+  : first(first_),
+    num_first_symints(-1),
+    num_first_tensors(-1),
+    second(second_),
+    num_second_symints(-1),
+    num_second_tensors(-1) {}
+
+std::vector<c10::SymInt> ChainedViewFunc::get_symints() {
+  auto symints = first->get_symints();
+  num_first_symints = symints.size();
+  auto second_symints = second->get_symints();
+  num_second_symints = second_symints.size();
+  symints.reserve(symints.size() + num_second_symints);
+  symints.insert(symints.end(), second_symints.begin(), second_symints.end());
+  return symints;
+}
+
+void ChainedViewFunc::set_symints(const std::vector<c10::SymInt>& symints) {
+  TORCH_INTERNAL_ASSERT(num_first_symints > -1);
```

Comment:
Not the way it's exposed now. The setters and getters are used by the impl of `_view_func()` / `_view_func_unsafe()`, so if the setter is somehow called without calling the getter, there's a bug there.

I'm happy to change this to a standard `TORCH_CHECK` if we want to avoid the error type here being dependent on the current use case.

Reply:
I'm fine with the argument that these methods should ONLY be used through _view_func